Revolutionaries 2.0

FairlySadPanda opened the pull request:
# Overview

Revolutionaries is a classic Space Station 13 game mode, updated and revised for maximum chaotic fun. A cabal of Head Revolutionaries, specialist infiltrators from the Syndicate, have sneaked aboard the Space Station. Their job: deliver a functional space station into Syndie hands.

To do this, the Head Revolutionaries use their own guile like other traitors, but they also have access to tools that alter the loyalties of defenseless crew members. These tools provide a way of the Head Revolutionaries turning the staff against the station, creating a team of loyal operatives or a disposable human tide that can overwhelm the station.

Like all good political insurrections, the drama is in the betrayal.

deathride58 commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,275 @@
+
+### Being a Rev has a tell, similar to some old SS13 conversion modes.
+
+Being a Rev means you look slightly dazed and confused, which can be hidden by hiding your eyes. This effect will also be applied to some drugs in game like Mindbreaker Toxin.
```
This is extremely gameable, as it strongly encourages shift-clicking everyone in sight for the purpose of validhunting, and discourages revs from trying to blend in

CrigCrag replied:
"which can be hidden by hiding your eyes."

Aexxie replied:
There needs to be SOME way to confirm a headrev.

FairlySadPanda replied:
See the Role Outline for Revolutionary:

```Converted Revolutionaries have an obvious trait that can be inspected:

“They appear slightly dazed and confused”.

This state is shared with imbiding some contraband chemicals, like Mindbreaker Toxin: being dazed and confused itself is not evidence of the round type being Revolutionaries.```

You are not valid if you're seen "dazed and confused" but it should be suspicious, and you can prevent it by wearing eye protection.

FairlySadPanda replied:
Please read the last sentence of the above post as "please, God, somebody think of a better idea than eyes, because I'm 99% sure I've stolen that idea from a Cultist implementation on TG or Goon and it wasn't great there"

deathride58 commented:
Honestly the vast majority of this is *really* good. Revs is pretty much just Katamari Damacy but with greytiders. An implementation of this proposal would give revs a far more unique feel that does a much better job at playing into the paranoia aspects of station antags.

Ratonero500 commented:
I think its a really cool concept,the only way i see of "metagaming" revs is noting that suddenly a lot of people on station are wearin sunglasses or masks,but i like all the other concepts.

mirrorcult commented:
"revs arent allowed to do antag stuff unless someone is 'defending a head'" does not sound like a recipe for less admin involvement, that sounds like a nightmare

mirrorcult commented:
same type of weird situationally-dependent half-antag status that led the various iterations of gangs & thief on tg to be terrible

tday93 commented:
Possible mechanical way to help prevent murderhobo-ing would be to have killing someone cause a substantial station-wide negative effect on the killer's faction's influence. Possibly even greater effect for gibbing. Revs killing = less rev influence, non-revs killing = more rev influence.

For situations where that's hard to track, it could be coupled with a more mild "martyrdom" bonus for one's own faction if you're killed.

Stealthbomber16 commented:
> Addition of Influence as a tracked player stat, and associated gameplay features
Influence is a value that each player carries. MindShielded players are always value 0. Influence is raised by a few routes (exposure to Head Revs, propaganda and one or two restricted drugs like Mindbreaker Toxin).

I'm not sure that this is a better solution than just flashes, though. This feels very metagamable and abusable. A problem I've seen during rev rounds is people flocking to headrevs to get converted. In this case, they'd just stand near propaganda. I'm not sure I buy the strengths of this new system.

I like the idea of the round ending when revs capture the bridge with AA. That is a good, satisfactory ending to a rev shift. I don't think some of these structural changes around it would work well, though I see the vision for the most part.

FairlySadPanda commented:
> "revs arent allowed to do antag stuff unless someone is 'defending a head'" does not sound like a recipe for less admin involvement, that sounds like a nightmare

I'll need to reword this section I think. Rev antags should work like normal antags; you have a goal but "don't be a dick" is still the overriding rule, if that makes sense :/

mirrorcult commented:
when "dont be a dick" is the overriding rule you're essentially forcing admins to figure out "whos being 'a dick'" in every single ambiguous scenario instead of having it be handled mechanically

Hmeister-fake commented:
> Possible mechanical way to help prevent murderhobo-ing would be to have killing someone cause a substantial station-wide negative effect on the killer's faction's influence. Possibly even greater effect for gibbing. Revs killing = less rev influence, non-revs killing = more rev influence.
> 
> For situations where that's hard to track, it could be coupled with a more mild "martyrdom" bonus for one's own faction if you're killed.!

This would work very well imo, amazing PR eitherway.

mirrorcult commented:
there are lots of good bits and pieces mechanically here but any implementation that seeks to restrict what revs are allowed to do by just telling them "well dont go too far" and expecting admins to disambiguate what "too far" means situationally is not one that is going to work

Aexxie commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,275 @@
+
+Converted Revolutionaries have an obvious trait that can be inspected:
+
+“They appear slightly dazed and confused”.
```
With how many mentions there have been with metagaming, I feel it necessary to point this out: if someone is seen with that inspect status, they WILL be metagamed by someone at some point. It is pretty much inevitable. It is the same case of how it is now; security sees a passenger with a flash, they immediately call revolutionaries. Even if a security member did not call a revolution early (based on the sighting of this status), they certainly would if they see many people with the same status.

To address it being hidden: it's a neat idea, but that alone is not going to stop metagaming from occurring.

FairlySadPanda replied:
I agree; it's currently the best idea for a drawback to being a Rev (that you're identifiable as valid by Security) I can think of.

There needs to be some drawback to being a Rev, otherwise people will go "oh, it's a Rev round, I'll be safe if I stand next to this poster then wander over to that particularly sus clown". Being valid is the best idea I have so far.

deltanedas replied:
not just by security but literally the entire station

FairlySadPanda commented:
> when "dont be a dick" is the overriding rule you're essentially forcing admins to figure out "whos being 'a dick'" in every single ambiguous scenario instead of having it be handled mechanically

That's fair. My view is that converted Revs should be under tighter rules than normal antags, as to give a set of people (potentially with very low experience playing the game) RR rights is a very bad idea, even if this refactor is intended to reduce the number of Revs in general.

I can consider some other mechanical way of making Revs not want to greytide the station; I'd be interested in hearing ideas.

Aexxie commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,275 @@
+
+## Core Mechanics
+
+### Propaganda
```
Really fun idea, but again, like above, will definitely be "metagamed." If you are walking around and see posters to join a revolution, you are going to call a revolution. I daresay at this point that this is not metagaming because it has a completely justified IC reason. I cannot see any reason why these posters would be used outside of a revolution, making them completely obvious when there is one.

FairlySadPanda replied:
Misdirection by other antags and "funny comedy" by clowns and mimes aside, this is a fair point.

SirChopwood replied:
Anything used as rev branding should be something that (at least at the start) can be masked as normal non-rev branding. That way it cannot be instantly meta'd as Revs. Could add plenty of options for service staff to add other propaganda such as TV presentations.

SirChopwood replied:
Still think we need more propaganda than posters, other methods that arnt as overt. Could be as suggested on the discord as some sort of code phrase / chant the HRs can call out to cause influence gain, for example.

deathride58 commented:
Honestly it does make sense for antags to remain consistent in being a license to grief; restricting converted revs via rules would make for a pretty confusing experience, especially for new players. It'd be better for extreme actions to be discouraged through means other than rule and hoping they're enforced (there's already really good discouragement in the form of Security. Plus, not having a good indicator of who's a rev or not means it's much harder for convertees to just blindly attack non-revs like revs currently tend to do, in turn meaning a high likelihood of friendly fire). Additionally, rev heads do have good reason to prioritize quality over quantity for conversions with this proposed rework (as stealth is actually meaningfully viable with this proposed rework), unlike current revs where quantity is vastly preferred (As all that matters in current revs is that heads die somehow. More bodies means overcoming resistance in making that happen).

Aexxie reviewed (commented):
Here are a few things I would like to note from both technical and gameplay standpoints. Clearly it has a few things that need to be worked out, but overall is a good idea and a much needed change. I would discuss this more with people more knowledgeable than I, it is completely possible that I missed some major technical issue in this review (and, of course, it is always good to get more viewpoints anyway.)

I think the major thing I noticed with this is the following: while this eliminates the old methods of metagaming to identify a revolution, it simply introduces new ones that work much in the same manner. This needs to be addressed, of course, because otherwise the entire point of this refactor would be meaningless.

That aside, I want to end on a positive note rather than leaving mostly negative comments. I think this is a really really cool concept, and yeah its gonna need some work, but change is absolutely needed and, quite frankly, I welcome them. An overhaul of revolutionaries is long overdue. Keep discussing with people, I'd like to see a finished product someday!

Emisse commented:
revs should not be under tighter rules than other antags. there is precedent for this, anything that should be stipulated by rules beyond "dont murder your teammates" should be enforced mechanically

FairlySadPanda commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,275 @@
+
+Being a Rev means you look slightly dazed and confused, which can be hidden by hiding your eyes. This effect will also be applied to some drugs in game like Mindbreaker Toxin.
+
+### Killing the Heads is now not required for the Revs to win, instead requiring taking the Bridge and using all Head ID cards on the Command Console and defending the Bridge for a period of time.
```
Good comments. One alternative for the cards I was considering was a retinal scan of each of the recognized Heads (meaning you can gib, murder or capture a Head and drag 'em to the Bridge) but I went for IDs as it's the universal security/ID chit.

For the capture-the-point problem: I'm sort of fumbling around with what exact terminal to use. One "uber terminal" that reflects the core Centcomm uplink is probably what would have to be needed, which would probably be made of completely indestructible materials. Like a couple of other points so far it makes me :/ as it is a bit video game-y

SirChopwood replied:
What if for the round to end every "Head" (or someone with that head's access) must activate some thing on their office's computer terminal. Once a majority of the buttons are on, the bridge comms console then has a final button to press to have the station succeed from NT. This would mean the revs need to hold territory in the form of each head office and the bridge and if they allow defending loyalists to take over a majority again and disable them, they must take it back. 

Revs Start > Take control of over 50% of Head Offices > Activate Computer Button > Take control of bridge > Activate comms console button > Win

FairlySadPanda commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,275 @@
+
+Your implant gives you access to the Syndicate radio channel innately. This can be changed to a unique Revolution radio channel, but given Nukies, Traitors and Revs are exclusive to each other, it should be fine to just use the Syndie channel.
+
+### Influence
```
Yep that's the idea. I'll add that as notes with maybe an image demonstrating it

Aexxie commented:
> What about a mental shock/morale factor when being exposed to a dying loyalist? If the Head Rev stays near their flock, this debuff is negated, but a lone Rev going out murdering people will flip back.

I like the idea of grief, but this would curb a lot of strategic liberty HRs could have (In example, splitting people off into teams.)
This may work if the grief reduction was relative to the amount of revolutionaries you have nearby.

mirrorcult commented:
> # Amendment Proposal
> 
> What about a mental shock/morale factor when being exposed to a dying loyalist? If the Head Rev stays near their flock, this debuff is negated, but a lone Rev going out murdering people will flip back.

implement mood :godo:

EmoGarbage404 commented:
just gonna comment that this document is quite lengthy and the actual changes being made do not feel super clear. This isn't a statement on the quality of anything proposed, rather just noting that it's hard to determine what exactly is being proposed and the specific mechanics of how those things would work.

Aexxie commented:
> just gonna comment that this document is quite lengthy and the actual changes being made do not feel super clear. This isn't a statement on the quality of anything proposed, rather just noting that it's hard to determine what exactly is being proposed and the specific mechanics of how those things would work.

Despite the "no its not a maintainer meeting" in contrib vc, its essentially a maintainer meeting. If youre interested, we are discussing it.

FairlySadPanda commented:
Agree with EmoGarbage that this is a lengthy doc and I'll need an executive summary at the top. This is a full game mode rebuild and by necessity is more complex than normal.

I've done a signficiant rework of numerous aspects of this PR:

1. Emphasised the paranoia aspect in the overview
2. Simplified Rev victory
3. Removed "pls no grief" admin rule
4. Outlined how Conversion works
5. Reworded the MindShield situation to be clearer that it's intended that you can run out if you recruit like five Security staff or something
6. Outlined the basics of gaining/losing Influence
7. Added poster creation and placement as traitor tasks
8. Added that the Clown can make their own funny posters, because that's just fair
9. Noted how to destroy the effect of poster
10. Defined the loudspeakers being destroyed as a skill issue
11. Removed Revs having a tell in favour of under-suspicion staff being subjugated to the Interrogator, a new electic chair
12. Be more specific in the role of the Centcomm Uplink Terminal and the Head Of Staff IDs in their respective sections
13. Outlined exactly how winning works. You need ONE Head Of Staff ID to start a long countdown, and each additional ID shortens that countdown. so you are encouraged to get as many as you can find, but you don't need all of them to try and force a win.

FairlySadPanda commented:
Just to stress in its own post: this mode is a rebuild but not a deliberate replacement of Revs 1.0 and I'd intend to code it to sit alongside that mode so that it can be feature toggled between; this rewrite _will_ have players break it over its knees repeatedly and it's not fair to punish said breaking with "wow, we've now disabled Revs whilst we do a rebalance, enjoy less round types!"

Hotfixes to the current mode are cool, like making flashing a bit better or having the game _sodding end when all the Heads are dead please god_

FairlySadPanda commented:
TODO 

THE PSYCHOLOGIST SHOULD BE ABLE TO READ SOMEONE'S INFLUENCE STATE

MY GOD THAT'S FUCKING OBVIOUS WHAT WAS I THINKING

K-Dynamic commented:
I was thinking introducing chem components like mindbreaker toxin to influence people (including revs and non-revs)

Ilya246 commented:
> TODO
> 
> THE PSYCHOLOGIST SHOULD BE ABLE TO READ SOMEONE'S INFLUENCE STATE
> 
> MY GOD THAT'S FUCKING OBVIOUS WHAT WAS I THINKING

shouldn't be innate, should be shown by a type of glasses

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+
+2. Input the cards and click the button to start the unlinking of the uplink.
+
+3. Wait some number of seconds for the uplink to be disabled. This alerts the entire station over comms that the Revs are doing this: I hope you have fortifed the bridge!
```
This all seems like a much better way to end the game. Provides crew and revs with a clear goal and space to play around. The use of extra head IDs to speed up/slow down the countdown seems like a perfect tug of war.

SirChopwood replied:
Only potential issue i CAN see is that it forces the bridge to become a battleground in every instance. Maybe discuss ideas for alternative locations to have this occur, could be any of the head offices so that, for example, revs setup in Medbay could work from CMO's and its then a medical rebellion.

FairlySadPanda replied:
At the moment I'm focussing on the Bridge alone so there's a three-step structure to a revolution round: the plotting phase with a small number of Revs, the stealing of Head IDs which is the mid-round goal and then the attack on the Bridge as the end game.

Depending on how this mode works in practice then I could see allowing for the flag to be captured from the various head offices, but I'd worry that'd over-encourage one-department scheming, like a Rev round where the Revs just take over Science and fortify it ridiculously for two hours.

Jezithyr replied:
At the end of the round there is likely to be a battle anyways and I think it's far better if it's around the bridge rather than just the evac shuttle getting bombed. Using the bridge as a more central objective point helps decrease the focus of antags from the evac shuttle at round end which will reduce shuttle bombings.

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+
+This terminal is the core security uplink between the space station and the rest of the universe (in fiction). It can send announcements and notifies who is the current chain-of-command.
+
+This console is indestructable, protected by military-grade force fields, immovable and powered by a nuclear engine that will outlast the star it's anchored to.
```
Is this not the existing Communications Console that's in the bridge and cap's office?

FairlySadPanda replied:
There's multiple comms consoles but the one on the bridge needs to be upgraded to be the "special" one that is the logical centre of the entire space station, as it's the flag that the Revs want to capture

SirChopwood reviewed (changes requested):
Great draft, needs work for some balance aspects and clarification on certain aspects to prevent it being a *different* metagamed mess. But that's all part of the fun :)

Vonsant commented:
Hmmm... Is it possible to implement that the head of the revolutionaries can "broadcast" his influence through the TV if he is shown on it and says something?

FairlySadPanda commented:
> > TODO
> > THE PSYCHOLOGIST SHOULD BE ABLE TO READ SOMEONE'S INFLUENCE STATE
> > MY GOD THAT'S FUCKING OBVIOUS WHAT WAS I THINKING
> 
> shouldn't be innate, should be shown by a type of glasses

For the psychologist I think it'd be funny if it's visible if someone's lying on, like, the psych's sofa. IDK if that exists yet as I know little about that role.

 I think that there's probably going to have to be some sort of eye-wear that spots rev propaganda but if there's clothing that shows off Influence levels that just means it gets confiscated by the HoS within microseconds of spawning.

FairlySadPanda commented:
> Hmmm... Is it possible to implement that the head of the revolutionaries can "broadcast" his influence through the TV if he is shown on it and says something?

I'm not going to touch the Reporter/TV game as part of this (and probably won't touch Psychology either) as they're niche roles that don't affect most stations. It's a good idea though, especially for RP.

FairlySadPanda commented:
> I was thinking introducing chem components like mindbreaker toxin to influence people (including revs and non-revs)

Mindbreaker Toxin is in as an Influence-generating drug.

I could consider a pacifying drug that Sec could administer, too...

HoidC commented:
> > Hmmm... Is it possible to implement that the head of the revolutionaries can "broadcast" his influence through the TV if he is shown on it and says something?
> 
> I'm not going to touch the Reporter/TV game as part of this (and probably won't touch Psychology either) as they're niche roles that don't affect most stations. It's a good idea though, especially for RP.

nah hear me out right spitballing here

what if one of the methods to gain influence is to hear certain words… code words, if you will? That the head revolutionaries get (randomised)? So if you hear a guy with really repetitive speech patterns that might be an indicator that he’s a revolutionary

FairlySadPanda commented:
OK, all reviewers so-far are re-requested

Can I get some specific to-dos or blockers for this, please?

My next job otherwise is to mock up the art and go through the to-dos for actually implementing this, but I need to know if the fundamentals are OK.

As a top level plan I can see a few clear milestones:

MILESTONE 1: WIN CONDITION
* Add the Centcomm Uplink Terminal to all maps.
    * Sprite
    * UI
    * Sound effects
    * Prototype, Component and System code
* Update spawned Head ID cards to be the keys to the Uplink terminal
    * New UplinkKey component attached to the Head ID cards on spawn (even late-spawn)
    * Make it obvious that a card is an uplink key card via text when examined (IIRC head IDs are already unique)
* Change win-condition of Rev rounds from murder-all-heads to defend-the-bridge
    * This is behind a cvar feature toggle
    * Max length of timer is a cvar
    * Min length of timer is a cvar
    * Each card inserted sets timer to the product of lerp(max, min, number of inserted cards/total cards spawned)

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+Conversion works as follows:
+
+1. The Revolutionary Implant grants the owner the ability to see the Influence level of those around them.
+2. If the Head Revolutionary sees a crew member with a high enough Influence level, they can activate their implant's ability and click on the target.
```
Are there any prospective stats or demos for exactly how this will work? Could end up being no different to flashes or too powerful one way or the other so certainly needs prototyping.

FairlySadPanda replied:
To-do for tonight is to make a prototype of this

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+
+Flashes are metagameable, confusing for newbies, hard to balance, and enforce too much reliance on manufacture by Science by the Revs for effective round flow. They also have no narrative justification beyond it being an Ur-mode from the dawn of SS13. There are people playing SS14 that are younger than the only good _Men In Black_ movie.
+
+### Addition of Influence as a tracked player stat, and associated gameplay features
```
Is there going to be a way for a player to know their influcence level. Something vague would be ideal to keep away too much metagaming but equally you dont want to heavily impact players by something they have zero feedback on.

chromiumboy replied:
Unless rev posters and randomly found about the station, having your influence change would instantly tell you what game mode is running

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+
+### Influence ticks up when exposed to sources and slowly ticks down when not. There is a max Influence level.
+
+Addition of NT and Revolutionary posters and propaganda, that both sides, and traitors, can put up and take down across the station.
```
Could potentially tie into the proposal for tourists too, giving them another task that "baits" antag meta'ing

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+
+Posters can be be destroyed by spraying them with paint or tearing them down.
+
+### Addition of Loyalty Loudspeakers in Sec
```
More themeatically might be to say these are "Mindshield Emitters, An experimental device utilising high power transmitters to project a small field of thought manipulation blocking rays similar yet weaker compared to existing Nanotrasen implants." or similar spess tech.

FairlySadPanda replied:
Yeah, the fluff can be revised depending on feedback. I like the "funny" aspect - calming Revs down via a hokey solution like smooth jazz - but it depends on the implementer to come up with a specific idea and get it merged.

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+
+These Loyalty Loudspeakers are audio emitters that reduce a player’s Influence level, even if they’re converted. If a player who is converted de-converts, their Influence immediately drops to zero. This is a new machine that requires quite a bit of power to run. A button is added to the Warden’s office to disable and enable all loudspeakers. The loudspeakers play calming, smooth jazz.
+
+These Loudspeakers cannot be made on the station. One of the key jobs of Sec is to ensure Sec remains safe; if it's been bombed and the Loudspeakers are inoperable, the Revolution is succeeding.
```
I'd argue allowing more of these to be constructed about the station to be fine as it gives the non-revs a midterm goal while waiting for the Head Revs to reveal themselves. Means that sec can fortify another department if they have trust in them not being revs, but doesnt completely nullify the ability for said dept to rise up against them later. Would also prevent revs just suicide bombing sec 2 mins in to remove the speaker and make it impossible to win for Sec.

FairlySadPanda replied:
I can see the argument. The problem is that "IT'S REVS, COVER THE STATION WITH ANTI-REV MECHANICS" is a current problem with the round type.

So what we want is a limited number of auras of de-conversion that Sec can control the placement of and that can be replenished if Sec loses them, but that Sec cannot spam.

SirChopwood replied:
Yeah imposing soft limits on how many they can use is what i'd encourage to do. Makes it less gamey as they cannot spam them everywhere but also allows for repairs or expansions if the station is doing particularly well. Also encourages sec to work with other departments and not just glare at them all from the brig.

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+
+The Interrogator is a high-power-draw device that lives in Sec. This machine does three things:
+
+1. Minorly fries the person placed inside it. `Your hair stands on end!` If you're a Diona, you are set on fire. `The lightning ignites your limbs!` It does no damage to Slimes. `You feel tingly`.
```
Funni electric chair, yes. However this does also not help with security trying to maintain order when they're electrocuting and frying their fellow staff. The interrogator simply being some sort of mind probe that could do something like radiation damage might hold a similar effect without it looking quite as bad. Also if slimes cant be shocked logically they cant be tested on it either?

FairlySadPanda replied:
"However this does also not help with security trying to maintain order when they're electrocuting and frying their fellow staff."

This is the point and a reason for the device not to be abused. Like the loudspeakers this will be flavour (I thought of making a reference to the box in Dune that is used to test if someone panics under pressure or not)

SirChopwood replied:
Despite good intentions i dont believe that's how it'd play out. You'll instead just have security taking a more extreme hardline of electrically torturing anyone they suspect of not being an extreme loyalist.

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+
+1. Minorly fries the person placed inside it. `Your hair stands on end!` If you're a Diona, you are set on fire. `The lightning ignites your limbs!` It does no damage to Slimes. `You feel tingly`.
+2. Reads their Influence level.
+3. Causes the lights to flicker station-wide.
```
Please no... Engineering cause that enough as is

FairlySadPanda replied:
The lights flickering is a reference to it being an electric chair - the high power draw causing the town's lights to flicker is a common motif

SirChopwood replied:
True but should leave that to the power system as is, if its overloaded then things power down. No need to artifically cause outages.

Jezithyr replied:
It's not really causing an outage so much as creating a tell for the rest of the crew that the chair is being used. This could also be tied into the influence system, where overusing the chair results in the crew's influence dropping

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+
+This allows Sec to ensure anyone they're releasing from prison has a low influence level.
+
+It can be emagged to dramatically increase the power draw, EMPing the station for a period of time, and instantly frying any poor sod put inside it.
```
Fine but again wouldnt like it to be too integral to the whole station/gamemode. Having it simply be an extra tool with small benefits and drawbacks in the right uses is good. EMP should only be to a local area around it, not the WHOLE station.

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+
+This removes one half of the round-removal problem and gives the Revs a clear final target that works a bit like the Nukie bomb.
+
+To compromise the terminal, the Revs need access to a Head Of Staff ID. Any will do: using them on the terminal allows for the Uplink to be disabled after a timer.
```
How does one disarm such event?

SirChopwood replied:
Also might suggest needing something more than one head-level ID and the bridge because that'd still be a quick way for any traitor to fuck with the WHOLE station or nukies to deviate from their actual objective? Perhaps some flavourtext about the HR implant being what enables the process and the ID only being needed to authorise it.

FairlySadPanda replied:
Maybe. I do want the Revs to be able to end the round if a HR is alive somewhere (but captured, maybe) so it'd be any Rev able to do it, but that might be metagamey.

I'll break the round ending mechanic into its own PR so it can be discussed, as it's the first major step and is already a rework to the mode ("Revs 1.5" if you will)

SirChopwood replied:
If the head rev is captured that should naturally cause it to be a revolutionry failure. They should be working their hardest to get their leader back before they're executed for their crimes. Having non HRs get game ending power would make it all to easy to be abused.

SirChopwood commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+
+### Crush The Mutiny
+
+Once you’re confident it’s a Rev round - the HoS being missing or his ID being stolen is a bit of a clue - you need to find out who it is who’s been doing all the converting and stop them. That’s easier said than done - even de-converted revolutionaries might not know who the ringleaders are, but arresting and interviewing them after deconversion is a good place to start.
```
The lack of ANYONE knowing who the HRs are certainly helps the many cases where people have simply leaked who they are to sec

FairlySadPanda replied:
That's the intent. Sec can keep the mutiny down but needs to round up the ringleaders by doing their job, or reacting well when the Revs make a play.

Because the powers are via an implant, an empty implanter can be used to ID a head rev if they get caught.

SirChopwood reviewed (commented):
Very good update on the docs and ToDo. Just a couple more points of specifics still to do but feels so very nearly there now. Specifically for Mockups/Demos, Player Influence Feedback, the Loudspeaker/EChair and Influence Sources (Propaganda)

FairlySadPanda commented:
TODO: Show off conversion mechanic in a prototype, break out round-ending mechanic into its own doc so it can be discussed/merged without the baggage of the conversion changes

Rainbeon commented:
Honestly? I like this a lot, no randomness, lots of dynamism, mind breaker is a great touch. My only real gripe is the complete severance of flashes from the mode when they're very, very iconic for the mode. It would be nice to still see them in some way even if minor but if they can't be fit into the big picture mechanically I'd understand too.

SirChopwood commented:
Flashes will still exist, and may even see more sec use as they are still blinding tools. However they are not longer the be all and end all of identifying revs. Gone are the days of 2 mins into the round someone getting flashed and HRs getting outed.

metalgearsloth commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,333 @@
+  - Boring stalemates and unclear round state, especially for the loyalist crew
+  - The endless feels-bad forced round-removal of discovered Head Revs and Heads Of Staff, sometimes for over an hour
+  - Paint-by-numbers round activities entirely focussed on flashes and MindShield implants.
+- A lengthy discussion at the 30th December 2023 Maintainer Meeting surrounding Revs, specifically flagging its problematic impact on Salamander/MRP
```
This shouldn't be used as a point here as it's not a part of the meeting.

OliverOtter commented on src/en/proposals/fairlysadpanda-revolutionaries.md:
```diff
@@ -0,0 +1,357 @@
+
+This implant does three things:
+
+1. Gives the HR innate access to talk on the Syndicate radio.
```
Giving Private radio comms is strong in my opinion. Giving them private communication without a trade of sorts feels over the top. For traitors, they need to use their TC, a limited resource, to get it. and even then hope that other traitors (if they havent found any yet) also get it. Having it start for HeadRevs for essentially free, can make snowballing a lot easier.

LankLTE commented:
Thank you for the document. After discussion with admins and maintainers, we feel that the current proposal struggles to actually fix many of the issues that exist in current revolutionaries while also introducing some issues of its' own. I’m closing the design document as the issues are fairly major, and we feel the proposal would benefit from being revisited rather than having specific issues addressed. Outlined below are what we feel are the main issues with revolutionaries- both currently and with this proposal- and how they could be addressed for any future proposal you may give. 
Metagaming - Current revolutionaries suffer a lot from being metagamed roundstart, revealing the antag and significantly shifting the way the crew plays. This proposal fails to solve this issue by introducing several other ways that the crew can metagame the existence of revolutionaries, such as, for example, catching someone putting up posters. Similarly, the head revolutionary being implemented through an implant further encourages the crew to metagame using implanters on most of their prisoners, which is similarly unfun for the revolutionaries.
Interesting counterplay - Security should have ways to counter revolutionaries that are fun for both antags and non-antags. Currently the "meta" is to spam mindshields as soon as revs are discovered, with little counterplay from revolutionaries. The current document seems to encourage security to just let people sit in cells while their influence goes down. This also does nothing to discourage security from just killing people should their speakers break or it be too inconvenient, which is also an issue with current revolutionaries. Having headrevs able to be deconverted with something as simple as an implanter isn’t particularly fun for either side as well. 
Willing conversion - The current revolutionaries have a large issue on the admin-side from players looking to get "free antag" status, essentially throwing themselves into the headrevs. There’s no specific solution to this, but something should probably exist to discourage willingly converting yourself, or integrating this into revolutionaries further and balancing around it. 
Influence - The influence mechanic abstract and unintuitive. Players don’t have any good way of understanding that they’re being affected by influence or how they’re being affected by influence. As a whole the opinion on influence varied, but there was at least a consensus that influence should be less of a hidden value and rather something easier for players to grasp should it remain the primary conversion method of revs. 

Other maintainers or admins may have more specific opinions, but I’ve tried to leave it to what was reached as a consensus rather than throwing ideas at you. Please feel free to ask if you have any questions about anything I’ve outlined here.

FairlySadPanda commented:
Good writeup. Considering this is a closed maintainer discussion I'll refrain from revisiting this; it'll be up to those at maintainer level to architect a better mode.

Jezithyr commented:
@LankLTE No this is not closed. Also, do not close design PRs in the future, you alone do not speak for the maintainers and other designers in this community. If you have issues with a design PR, please discuss it in the internal design channel before taking any action.

LankLTE commented:
Jezithyr missed the internal discussion (thanks, threads), but there was one previously which led to the reply I gave. We’re still going to be re-discussing this in a manner that hopefully won’t cause anyone to accidentally miss it and ensure this doesn’t happen again. 

> Considering this is a closed maintainer discussion I'll refrain from revisiting this

I’d also like to not let this discourage you. The closed discussion was simply to ensure that admins & maints are on the same page as to what the issues with revs actually were, but that doesn’t mean we aren’t open to ideas to fix it or reviewing more documents in the future. Of course if you’d still like to focus on fixes and such anyways, feel free.

FairlySadPanda commented:
> > Considering this is a closed maintainer discussion I'll refrain from revisiting this
> 
> I’d also like to not let this discourage you. The closed discussion was simply to ensure that admins & maints are on the same page as to what the issues with revs actually were,

No, I understand that; from experience at work I know that there's a lot of hurdles that have to be overcome to get a large change through design, and it needs stakeholder buy-in and time to build consensus, and that's challenging when it's done via PR Hell.

I think one thing to improve this process might be to limit who is able to review/comment on design docs - a lot of the chatter here has been from other contributors. It's maintainer/admin opinions that matter but it's hard to focus on specific feedback.

FairlySadPanda commented:
After the chatter tonight in a thread on the Discord to go over folks' thoughts I've got a couple of actions to do:

1. The "win condition" half of this proposal is waaaaay less controversial than the conversion part and should be broken out into a PR of its own.

2. The Influence mechanic needs a good tree shake - it needs to be fully justified and well-explained.

In general the common agreement is that there's improvements needed to the mode, but the background statement I wrote at the top of the PR is not a consensus. There's been some debate about the exact point of the game mode and whether enforcing round removal is a problem or not.

There are some aspects of feedback that are essentially wrecking amendments; they're not able to be fairly actioned by anyone without a fundamental rewrite of SS. In particular:
1. Metaknowledge. It will always be impossible to police and control knowledge of game systems with regard to effective antag and crew strategies once the round is known. There's mitigations that can be done - for example, the removal of a tell that someone's been Rev'd means validhunting is less good against Revs. But as long as how Revs convert is public knowledge, players will make actions based on their own view of the correct way to play the game - either to power game, be casual, or RP - and this simply cannot be controlled by a design doc. A good example of this is the nuke disk and the problem that used to happen in the days of yore on Goonstation where captains would throw the nuke disk and their antique raygun out the airlock as they were antag targets. Fundamentally you can't go "you can't make the nuke disk a target, the captain will throw it out an airlock", because you're basically locking yourself in to only having very specific types of game mode involving secret traitors or murderhobos.
2. Self-antagging. You cannot fundamentally stop someone from going "oh it's Revs? Time to get my antag juice!". You also can't stop someone from ignoring their R indicator when flashed, or ignoring their Traitor objectives. That's not how Space Station works. And hell, the dream in this game mode is voluntary conversion anyway, because it's cool RP with Head Revs spewing polemics and whatnot.

In terms of Influence it will need workshopping. I really like the idea, but making paranoia fun and fair for everyone is gonna be hard and will need a lot of selling, and posters are not really the most evocative and fun space thing to do as a feature. "oh boy I get to put up posters" is funny to me as someone who has visited Paris just before their elections and just looking at the _total nonsense_ of posters everywhere, but most people don't have that perspective. So for Influence I will need to make it sound more fun when explained.

Humbledwanderer commented:
🤩💀🤩
